app/src/main/java/com/example/epamandroid/MyService.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.epamandroid
+
+import android.app.IntentService
+import android.content.Intent
+import android.os.Build
+import android.support.annotation.RequiresApi
+import android.widget.Toast
+import java.time.LocalDateTime
+
+class MyService : IntentService("MyService") {
+
+    private val myIntent: Intent = Intent(Constants.CUSTOM_ACTION)
+
+    private val colorList: MutableList<String> = ArrayList()
+
+    private var serviceStatus: Boolean = false
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onHandleIntent(intent: Intent) {
+        serviceStatus = intent.getBooleanExtra(Constants.SERVICE_STATE, false)
+
+        do {
+            Thread.sleep(350)
+            when {
+                serviceStatus -> myIntent.putExtra(
+                    Constants.BROADCAST_MESSAGE,
+                    colorList[(Math.random() * 100 % (colorList.size - 1)).toInt()]
+                )
+                !serviceStatus -> myIntent.putExtra(
```

Comment:
In your case can be `else` instead `!serviceStatus`

Reply:
@egorikftp done